File: 1. Two Sum/twoSum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> seenMap = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; ++i) {
            int needed = target - nums[i];
            if (seenMap.containsKey(needed)) {
                return new int[] { seenMap.get(needed), i };
            }

            seenMap.put(nums[i], i);
        }

        return new int[2];
    }
}

File: 1. Two Sum/twosum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let map = nums.map((val, indx) => [val, indx]);
  map.sort((a, b) => a[0] - b[0]);

  let lp = 0;
  let rp = map.length - 1;

  while (lp < rp) {
    if (map[lp][0] + map[rp][0] == target) return [map[lp][1], map[rp][1]];
    if (map[lp][0] + map[rp][0] < target) lp += 1;
    else rp -= 1;
  }
};
